File: bricks/savana_plugin/{{plugin_name.snakeCase()}}/__brick__/{{plugin_name.snakeCase()}}_ios/ios/Classes/{{plugin_name.pascalCase()}}Plugin.m
#import "SavanaPluginPlugin.h"

@implementation SavanaPluginPlugin

+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
  FlutterMethodChannel *channel =
      [FlutterMethodChannel methodChannelWithName:@"savana_plugin_ios"
                                  binaryMessenger:registrar.messenger];
  [channel setMethodCallHandler:^(FlutterMethodCall *call, FlutterResult result) {
    if ([@"getPlatformName" isEqualToString:call.method]) {
      result(@"iOS");
    } else {
      result(FlutterMethodNotImplemented);
    }
  }];
}

@end

File: bricks/savana_plugin/__brick__/{{plugin_name.snakeCase()}}/{{plugin_name.snakeCase()}}_macos/macos/Classes/{{plugin_name.pascalCase()}}Plugin.swift
import FlutterMacOS
import Foundation

public class {{plugin_name.pascalCase()}}Plugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(
      name: "{{plugin_name.snakeCase()}}_macos",
      binaryMessenger: registrar.messenger)
    let instance = {{plugin_name.pascalCase()}}Plugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformName":
      result("MacOS")    
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
